src/services/budgeting.rs

```diff
@@ -34,67 +34,70 @@ impl StepBudgets {
             goals.get_mut(&goal_id).unwrap().tags.push(Tag::Budget);
         }
         //For each budget add all descendants
-        for budget in &self.budget_id_to_budget {
-            let mut parents_to_go: Vec<String> = vec![budget.0.clone()]; //start with the goal that initiates the budget
-            self.goal_id_to_budget_ids
-                .insert(budget.0.clone(), vec![budget.0.clone()]); //add itself for when budget filler min-max need to be checked with budget
+        for (goal_id, _) in &self.budget_map {
```

Comment:
There is no goal_id in budget_map, right?

Reply:
Actually there is, that why I renamed it to avoid conflicts. Check this: https://github.com/tijlleenders/ZinZen-scheduler/blob/a003845f8895c8d156b5b084e6bf22d9f57fc981/src/models/budget/impls.rs#L74

Reply:
Ok I'll revert what I thought was a typo then...